02-peer-protocol.md

```diff
@@ -553,11 +553,24 @@ the other node then replies similarly, using a fee it thinks is fair.  This
 exchange continues until both agree on the same fee or when one side fails
 the channel.
 
+In the modern method, the funder sends its permissable fee range, and the
+non-funder has to pick a fee in this range. If it chooses the same value,
```

Comment:
Would be clearer if the non-funder was explicitly mentioned here instead of "it" (ambiguous context imo). 

> otherwise the funder will reply with the same value (completing after three messages) 

Also it isn't guaranteed that things complete in three messages if the funder chooses to pick a different value, right (responder wants a fee too low, maybe below the current mempool high water mark)? Since the non-funder may not like that advertised fee. 

Instead, seems this is just best effort, and if the responder doesn't immediately reply with the same value, then things fall back to the "old" method?

Reply:
> Instead, seems this is just best effort, and if the responder doesn't immediately reply with the same value, then things fall back to the "old" method?

Things never fall back to the "old" method? The funder can start over, so, yes, the text should be changed, but there's no fallback.

Reply:
```
- if there is no overlap between that and its own `fee_range`:
      - SHOULD send a `warning`
      - MUST fail the channel if it doesn't receive a satisfying `fee_range` after a reasonable amount of time
```

@Roasbeef the case you describe is a spec violation. The non-funder either chooses a fee inside the advertised `fee_range`, or doesn't answer (sending a warning once that has been added to the spec) potentially leading to a force-close instead if no progress is made.

We only fall back to the old negotiation when the response doesn't contain a `fee_range` (indicating that the receiver didn't support this new negotiation scheme).

Reply:
The flow from the node operator's point of view in that case is:

- use the API to close a channel, with an explicit `fee_range` that the node operator think makes sense
- if the peer accepts that, great!
- otherwise the node operator receives a warning, and can choose to either retry with a different `fee_range`, do nothing or force-close

In the future we may want to automate that retry, but based on our users' feedback, they would really prefer having a human in the loop to decide whether they should adapt their `fee_range` or not. Many have been disgruntled by the feerates chosen by estimators, so they're not interested in an automated retry here.